### ISSUE TITLE
rotating and moving a object(literally freeform object manipulation) with spacemouse instead 2d mouse would be amazing ;)

### ISSUE THREAD

**Issue by stjema** [microdee/UE4-SpaceMouse#77]

**Comment by AltonLord** [comment]
I think a hotkey button on the Spacemouse with a UI button backup would work great.  I think lmb would be too obtrusive as it is already ingrained in most people's workflow to use it like it is.

I gotta admit, that would be an amazing feature.

**Comment by stjema** [comment]
but the sussy part is unitys spacemouse for example has no google results in the last few years. only for 2013 etc.
seems like its not efficient workflow. people stop using spacemouse?

**Comment by microdee** [comment]
> In GrabMove mode the asset will be linked to your camera so you can take it with you and position it where you want.

this is already a feature in the engine, you can attach objects to your camera, or you can pilot them as they were the camera, spacemouse works with both

**Comment by Nihilianth** [comment]
> > In GrabMove mode the asset will be linked to your camera so you can take it with you and position it where you want.
> 
> this is already a feature in the engine, you can attach objects to your camera, or you can pilot them as they were the camera, spacemouse works with both

@microdee I only saw that feature being implemented for game mode. Is there also one for editor?

**Comment by microdee** [comment]
yes you press Ctrl+Shift+P when an object is selected and now you fly around with that object. To exit press the eject button on top right of the viewport. You can modify this shortcut with the editor keyboard binding, which can also accept and learn buttons from the SpaceMouse

**Comment by itognet** [comment]
Sorry guys for interfering. 
I was looking for exactly this. Transforming an object would be awesome! with a hold middle-mouse to activate SpaceMouse transformation and then be able either to move, rotate and scale an object.

**Comment by myboxi** [comment]
i want this , did u make it to work ? i was curious if my idea will works. so someone did it already? or i need to do it myself
